doc/devel/git_workflows.rst

```diff
@@ -0,0 +1,44 @@
+
+.. _git_workflows:
+
+Git workflows
+=============
+
+When reviewing contributions, we sometime refer to this page to offer more detailed git workflows.
+
+.. _git_rebase:
+
+Rebase
+^^^^^^
+
+If you want to rebase, the first thing to do is to squash all your commits into one, which will make the job easier.
+Make sure you are in the PR branch, then to rebase do::
+
+$ git rebase --interactive upstream/main
+
+
+Git opens the last commits you made in our terminal editor (often it's vim) and you need to follow the instructions in
+the file. Basically replace 'pick' by 'fixup' (or simply 'f') in all but the first commit (exit vim using `:wq` to
+write/save and quite). Then update your main branch from upstream, change back to the PR branch and do::
```

Comment:
I thought adding the exit vim clue would save a lot of people a lot of searching 😉 I saw something similar in the [GitLab documentation for interactive rebase](https://docs.gitlab.com/ee/topics/git/git_rebase.html#interactive-rebase) and I thought it was kind and helpful. I tried to keep it more concise and be clear it's just an example.

Reply:
I'd hate for people to read our instructions and come away with the impression that they have to learn `vi` before they can master a rebase.  If you want a generic vt100-compatible text editor `pico`/`nano` would be a _lot_ more suitable for beginners who are just slavishly following our instructions.

Reply:
I wholeheartedly agree. I don't think I implied that people need to master vim though?! At least on macs vim is the default editor, and I don't know how many beginners change it 🤷‍♀️  So likely the first time they run a rebase it would open in vim, and they might not know what to do. This means we need to add [instruction for mac users](https://unix.stackexchange.com/questions/501862/how-can-i-set-the-default-editor-as-nano-on-my-mac) to change the `.bash_profile` before they run rebase to nano to change their default, and/or a hint on how to exit vim. I am not sure what's the common default editor on windows and linux.